### PR TITLE
Fix migration

### PR DESCRIPTION
**Pull request by adrianschroeter**
hopefully also fixes package builds

### CODE REVIEW

**Comment by hennevogel** [comment]
What are you fixing with these changes? The removed keys are still needed?

**Comment by adrianschroeter** [comment]
On Dienstag, 7. Februar 2017, 05:07:28 CET wrote Henne Vogelsang:
> What are you fixing with these changes? The removed keys are still needed?

as written, they only existed in current structure.sql, but you don't get actually when
updating from former versions.

So migration test is failing. We either need to sync the structure.sql or write migrations
to come to the same result when updating from OBS 2.5.

Changed structure.sql as first attempt, since it is easier and it looks like they come from
broken development system.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by evanrolfe** [comment]
Maybe this is a good time to discuss why we are overriding rails' default rake task for dumping to structure.sql? So far I have heard a reason for this?

https://github.com/evanrolfe/open-build-service/blob/master/src/api/lib/tasks/databases.rake#L31

**Comment by evanrolfe** [comment]
If this is blocking you then I'm fine for this to be merged but please could you create a trello card for reviewing the overriding of the rake task to dump db structure so we can discuss it in the next grooming meeting?

**Comment by adrianschroeter** [comment]
https://trello.com/c/Kn8ymPv7/175-change-db-structure-sql-handling

**Comment by hennevogel** [comment]
@adrianschroeter so you removed key that where never in structure.sql? I don't get it, please lets discuss this offline

**Comment by evanrolfe** [comment]
@adrianschroeter thanks for creating the trello card :)

**Comment by hennevogel** [comment]
We have discussed this offline now. The changes included in this PR are how the structure.sql looks like when you db:migrate from OBS 2.5 to now. So there are differences and we don't know where they come from. Merging this will make migration forward work again and we will follow up with DB migrations to add/remove the things we need again afterwards.

File: src/api/db/structure.sql [hunk 1]
```diff
@@ -1359,6 +1361,7 @@ INSERT INTO schema_migrations (version) VALUES
 ('20141201135426'),
 ('20141202135426'),
 ('20141208135426'),
+('20141302101042'),
```
Comment (evanrolfe) [inline src/api/db/structure.sql]: This migration does not exist. @adrianschroeter do you have any idea why this was added here? Perhaps the migration only exists on production?

File: src/api/db/structure.sql [hunk 2]
```diff
@@ -263,7 +263,7 @@ CREATE TABLE `bs_requests` (
 
 CREATE TABLE `cache_lines` (
   `id` int(11) NOT NULL AUTO_INCREMENT,
-  `key` varchar(4096) COLLATE utf8_unicode_ci NOT NULL,
+  `key` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
```
Comment (evanrolfe) [inline src/api/db/structure.sql]: There is no validation for this constraint in the [CacheLines](https://github.com/evanrolfe/open-build-service/blob/master/src/api/app/models/cache_line.rb) model but I don't see this causing any major problems at the moment.

File: src/api/db/structure.sql [hunk 3]
```diff
@@ -315,6 +315,7 @@ CREATE TABLE `channel_targets` (
   `id` int(11) NOT NULL AUTO_INCREMENT,
   `channel_id` int(11) NOT NULL,
   `repository_id` int(11) NOT NULL,
+  `prefix` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
```
Comment (evanrolfe) [inline src/api/db/structure.sql]: This column is not referenced anywhere in the code so is not needed.

File: src/api/db/structure.sql [hunk 4]
```diff
@@ -365,7 +366,7 @@ CREATE TABLE `configurations` (
   `change_password` tinyint(1) DEFAULT '1',
   `hide_private_options` tinyint(1) DEFAULT '0',
   `gravatar` tinyint(1) DEFAULT '1',
-  `enforce_project_keys` tinyint(1) DEFAULT '0',
+  `enforce_project_keys` tinyint(1) DEFAULT '1',
```
Comment (evanrolfe) [inline src/api/db/structure.sql]: @adrianschroeter @hennevogel do we want enforce_project_keys enabled or disabled by default?

File: src/api/db/structure.sql (hunk 2 above)
Comment (hennevogel) [inline src/api/db/structure.sql]: I remember seing exception about this being too small...

Comment (evanrolfe) [inline src/api/db/structure.sql]: @hennevogel I've changed this here: https://github.com/openSUSE/open-build-service/pull/2848

**Comment by Ana06** [comment]
[(5) P4: Review changes made by PR#2652](https://trello.com/c/UryVv6Jf/298-5-p4-review-changes-made-by-pr-2652)

**Pull request merged**